### PR TITLE
Add Service Bindings Panel in Function view in core-ui

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add Service Bindings Panel in Function view in core-ui:
  - Add modal for creating ServiceBindingUsage to inject secrets from given Service Instance to Function
  - Add Panel in `Configuration` tab for managing ServiceBindingUsages of Function
  - Add relevant unit tests 

**Related issue(s)**
https://github.com/kyma-project/console/issues/1311

### CODE REVIEW

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/test/Configuration.ui.test.js [hunk 1]
```diff
@@ -10,13 +11,16 @@ describe('Lambda Configuration Tab', () => {
...
+          lambda={lambda}
+          sizeRef={emptyRef}
+          runtimeRef={emptyRef}
+          LabelsEditor={labelEditorMock}
+          formRef={{ current: null }}
+          refetchLambda={() => {}}
+        />
+      </MockedProvider>,
     );
     expect(component.toJSON()).toMatchSnapshot();
```
Comment (akucharska) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/test/Configuration.ui.test.js]: As mentioned before it is better not to use toMatchSnapshot

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/test/ServiceBindingsWrapper.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import ServiceBindingsWrapper from '../ServiceBindingsWrapper';
+
+describe('ServiceBindingsWrapper', () => {
+  it('Render with minimal props', () => {
+    const component = renderer.create(
+      <MockedProvider>
+        <ServiceBindingsWrapper lambda={lambda} />
+      </MockedProvider>,
+    );
+    expect(component.toJSON()).toMatchSnapshot();
```
Comment (akucharska) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/test/ServiceBindingsWrapper.test.js]: Don't use toMatchSnapshot if possible

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/test/ServiceBindingsService.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { ServiceBindingsService } from '../ServiceBindingsService';
+
+describe('ServiceBindingsService', () => {
+  it('Render with minimal props', () => {
+    const component = renderer.create(
+      <ServiceBindingsService lambdaName="foobar">
+        <div>foobar</div>
+      </ServiceBindingsService>,
+    );
+    expect(component.toJSON()).toMatchSnapshot();
```
Comment (akucharska) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/test/ServiceBindingsService.test.js]: Don't use toMatchSnapshot if possible

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/test/ServiceBindings.test.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    const component = renderer.create(
+      <MockedProvider>
+        <ServiceBindings
+          injectedServiceInstances={[]}
+          notInjectedServiceInstances={[]}
+          refetchQuery={() => {}}
+        />
+      </MockedProvider>,
+    );
+    expect(component.toJSON()).toMatchSnapshot();
```
Comment (akucharska) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/test/ServiceBindings.test.js]: Don't use toMatchSnapshot if possible

File: core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+          {!createCredentials && secrets.length ? (
+            <FormItem key="existingCredentials">
+              <FormLabel htmlFor="existingCredentials">Secrets</FormLabel>
+              <select
+                id="existingCredentials"
+                value={existingCredentials}
+                onChange={e => setExistingCredentials(e.target.value)}
+                required
+              >
+                <option value=""></option>
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js]: Yes, empty option for choosing empty option :)

File: core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+      </FormItem>
+
+      {serviceInstanceName && (
+        <>
+          <FormItem key="createCredentials">
+            <Checkbox
+              name="createCredentials"
+              value="Create new Secret"
+              checked={createCredentials}
+              inputProps={{
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js]: I make it const outside the function :)

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindingsService.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+  const [deleteServiceBindingUsageMutation] = useMutation(
+    DELETE_SERVICE_BINDING_USAGE,
+  );
+
+  function handleError(
+    serviceInstanceName,
+    error,
+    mutationType = ACTION_TYPE.CREATE,
+  ) {
+    let actionVerb = 'creating';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindingsService.js]: Done

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindingsService.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+  async function handleDeleteServiceBindingUsage(
+    serviceBindingUsage,
+    refetchLambda,
+  ) {
+    const namespace = LuigiClient.getEventData().environmentId;
+    const serviceBindingUsageName = serviceBindingUsage.name;
+
+    try {
+      let response = null;
+      response = await deleteServiceBindingUsageMutation({
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindingsService.js]: In separate function or file? I want to have a connection to backend in separate file, so this service is only for aggregating functionalities.

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindingsService.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+      .showConfirmationModal({
+        header: `Remove Service Binding`,
+        body: `Are you sure you want to delete Service Binding from "${serviceBindingUsage.serviceBinding.serviceInstanceName}" Service Instance?`,
+        buttonConfirm: 'Delete',
+        buttonDismiss: 'Cancel',
+      })
+      .then(() =>
+        handleDeleteServiceBindingUsage(serviceBindingUsage, refetchLambda),
+      )
+      .catch(_ => {});
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindingsService.js]: I had a handle in `handleDeleteServiceBindingUsage` function, but ok, I remove it.

File: core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+  const [serviceInstanceName, setServiceInstanceName] = useState('');
+  const [envPrefix, setEnvPrefix] = useState('');
+
+  const [createCredentials, setCreateCredentials] = useState(true);
+  const [existingCredentials, setExistingCredentials] = useState('');
+  const [secrets, setSecrets] = useState([]);
+
+  useEffect(() => {
+    setValidity(false);
+  });
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js]: I change it to

```js
  useEffect(() => {
    setValidity(false);
  }, []);
```

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js]: Also I want have for it one file. Please see that this component is only form, so if I will split it to few files, then I must make logic with binding from child to parent or something more complex arch and now it's as simple as possible.

File: core-ui/src/components/Lambdas/Checkbox/Checkbox.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
+import { FormItem, FormLabel } from 'fundamental-react';
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef } from 'react';
+
+const getCheckStatus = (checked, indeterminate) => {
+  if (indeterminate) {
+    return 'mixed';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/Checkbox/Checkbox.js]: This component will be removed after bump react-fundamentals to 0.8.0. Actually version on master hasn't this component.

File: core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+              inputProps={{
+                style: {
+                  marginRight: '6px',
+                },
+              }}
+              onChange={(_, value) => setCreateCredentials(value)}
+            />
+          </FormItem>
+
+          {!createCredentials && secrets.length ? (
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js]: Here I checked only bool values, so if `secrets.length` will be false, then it return false not null value :)

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindings.scss [hunk 13]
```diff
@@ -0,0 +1,13 @@
+.lambda-service-bindings {
+  .fd-panel__body {
+    padding: 0;
+  }
+
+  .fd-panel {
+    margin: 0 !important;
+
+    .fd-panel__header {
+      display: none;
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindings.scss]: Removed

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindings.js [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+  const headerRenderer = () => ['Instance', 'Environment Variable Names'];
+  const rowRenderer = bindingUsage => [
+    <span data-test-id="service-instance-name">
+      {bindingUsage.serviceBinding.serviceInstanceName}
+    </span>,
+    renderEnvs(bindingUsage),
+  ];
+
+  return (
+    <Panel className="fd-has-margin-medium lambda-service-bindings">
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindings.js]: Done :)

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindings.scss [hunk 15]
```diff
@@ -0,0 +1,13 @@
+.lambda-service-bindings {
+  .fd-panel__body {
+    padding: 0;
+  }
+
+  .fd-panel {
+    margin: 0 !important;
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/ServiceBindings.scss]: I removed it :)

File: core-ui/src/components/Lambdas/LambdaDetails/LambdaDetailsHeader/LambdaDetailsHeader.js [hunk 16]
```diff
@@ -5,44 +5,65 @@ import LuigiClient from '@kyma-project/luigi-client';
...
   const notificationManager = useNotification();
 
+  function handleError(error) {
+    const errorToDisplay = extractGraphQlErrors(error);
+    notificationManager.notifyError({
+      content: `Error while removing lambda ${name}: ${errorToDisplay}`,
+    });
+  }
+
   const handleLambdaDelete = async () => {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/LambdaDetailsHeader/LambdaDetailsHeader.js]: Is very similar, coincidence of names.

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ConfigurationForm/test/ConfigurationForm.test.js [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+    const component = renderer.create(
+      <ConfigurationForm
+        lambda={lambda}
+        sizeRef={emptyRef}
+        runtimeRef={emptyRef}
+        LabelsEditor={labelEditorMock}
+        formRef={{ current: null }}
+      />,
+    );
+    expect(component.toJSON()).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ConfigurationForm/test/ConfigurationForm.test.js]: Done in all created files :)

File: core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (aerfio) [reply core-ui/src/components/Lambdas/CreateServiceBindingForm/CreateServiceBindingForm.js]: ok, fair enough

**Pull request merged**